### PR TITLE
feat/pdf viewer 2.0

### PR DESCRIPTION
**Pull request by talevy17**

### CODE REVIEW

File: packages/plugin-file-upload/web/src/createFileUploadPlugin.ts [hunk 1]
```diff
@@ -8,19 +8,42 @@ import type { CreatePluginFunction } from 'wix-rich-content-common';
...
       helpers,
       t,
       settings,
+      innerRCERenderedIn,
+      anchorTarget,
+      relValue,
+      uiSettings,
+      isMobile,
+      experiments,
     }),
```
Comment (danarish4) [inline packages/plugin-file-upload/web/src/createFileUploadPlugin.ts:36]: why did you pass all of this? 
` anchorTarget,
      relValue,
      uiSettings,
      isMobile,
      experiments`

File: packages/plugin-file-upload/web/src/createFileUploadPlugin.ts [hunk 2]
```diff
@@ -8,19 +8,42 @@ import type { CreatePluginFunction } from 'wix-rich-content-common';
...
     settings,
     t,
     defaultPluginData: DEFAULTS,
+    innerRCERenderedIn,
+    anchorTarget,
+    relValue,
+    uiSettings,
+    isMobile,
+    experiments,
     ...rest,
```
Comment (danarish4) [inline packages/plugin-file-upload/web/src/createFileUploadPlugin.ts:47]: I don't think this is necessary

Comment (talevy17) [reply packages/plugin-file-upload/web/src/createFileUploadPlugin.ts:47]: `anchorTarget` and `relValue` are indeed unnecessary, `isMobile` and `experiments` are needed for the settings modal and the plugin viewer

File: packages/plugin-file-upload/web/src/createFileUploadPlugin.ts (hunk 1 above)
Comment (talevy17) [reply packages/plugin-file-upload/web/src/createFileUploadPlugin.ts:30]: removed unnecessary fields.

File: packages/plugin-commons/web/src/Utils/mediaPluginsDataBuilders.ts [hunk 3]
```diff
@@ -76,7 +76,8 @@ export const dataBuilder = {
     return { ...componentData, src, error, tempData: undefined };
   },
   [FILE_UPLOAD_TYPE]: ({ data, error }, componentData) => {
-    return { ...componentData, ...data, error, tempData: undefined };
+    const fileData = { ...componentData, ...data, error, tempData: undefined };
```
Comment (danarish4) [inline packages/plugin-commons/web/src/Utils/mediaPluginsDataBuilders.ts:79]: are we still using tempData ? 
I think we should not support it anymore

Comment (talevy17) [reply packages/plugin-commons/web/src/Utils/mediaPluginsDataBuilders.ts:79]: we should and we will in the upload functionality revamp (:

Comment (danarish4) [reply packages/plugin-commons/web/src/Utils/mediaPluginsDataBuilders.ts:79]: revert file

File: packages/plugin-file-upload/web/src/file-upload-viewer.jsx [hunk 4]
```diff
@@ -21,31 +22,55 @@ const getNameWithoutType = fileName => {
...
+      settings: { adobeAPIKey },
+    } = nextProps;
+    const { componentData } = this.props;
+    if (!isEqual(nextComponentData, componentData)) {
+      validate(nextComponentData, pluginFileUploadSchema);
     }
     if (!nextProps.isLoading && this.props.isLoading) {
       this.switchReadyIcon();
     }
+    if (this.didPDFSettingsChange(nextComponentData, componentData) && adobeAPIKey) {
```
Comment (danarish4) [inline packages/plugin-file-upload/web/src/file-upload-viewer.jsx:67]: move to PDFViewer component

File: packages/plugin-file-upload/web/src/file-upload-viewer.jsx [hunk 5]
```diff
@@ -21,31 +22,55 @@ const getNameWithoutType = fileName => {
...
+      resolvedFileUrl: null,
+      resolvingUrl: false,
+      currentWidth: 0,
+      pdfStatus,
+    };
   }
 
   isInResizeRange = resizeWidth => this.fileUploadViewerRef?.current?.clientWidth < resizeWidth;
 
+  didPDFSettingsChange = (nextComponentData, componentData) => {
```
Comment (danarish4) [inline packages/plugin-file-upload/web/src/file-upload-viewer.jsx:51]: move to PDFViewer component

File: packages/plugin-file-upload/web/src/file-upload-viewer.jsx [hunk 6]
```diff
@@ -282,25 +307,70 @@ class FileUploadViewer extends PureComponent {
...
       <Tooltip
         content={this.isInResizeRange(resizeWidths.first) && componentData.name}
         tooltipOffset={{ y: 25 }}
       >
-        <div className={style} data-hook="fileUploadViewer" ref={this.fileUploadViewerRef}>
-          {viewer}
-          {this.renderAutoDownloadLinkRef()}
+        <div data-hook="fileUploadViewer" ref={this.fileUploadViewerRef} className={style}>
+          {shouldRenderPDFViewer ? (
+            <>
```
Comment (danarish4) [inline packages/plugin-file-upload/web/src/file-upload-viewer.jsx:353]: wrap with div with the classname file_upload_container

**Review by danarish4 — CHANGES_REQUESTED** [review]

File: packages/plugin-file-upload/web/src/file-upload-viewer.jsx (hunk 4 above)
Comment (talevy17) [reply packages/plugin-file-upload/web/src/file-upload-viewer.jsx]: done

File: packages/plugin-file-upload/web/src/file-upload-viewer.jsx (hunk 5 above)
Comment (talevy17) [reply packages/plugin-file-upload/web/src/file-upload-viewer.jsx]: done

File: packages/plugin-file-upload/web/src/file-upload-viewer.jsx (hunk 6 above)
Comment (talevy17) [reply packages/plugin-file-upload/web/src/file-upload-viewer.jsx]: done

File: packages/plugin-file-upload/web/src/file-upload-viewer.jsx [hunk 7]
```diff
@@ -282,25 +307,70 @@ class FileUploadViewer extends PureComponent {
...
+                  viewMode,
+                  showDownloadPDF: !disableDownload,
+                  showPrintPDF: !disablePrint,
+                  locale,
+                }}
+              />
+              {![PDF_STATUS.READY].includes(pdfStatus) && viewer}
+            </>
+          ) : (
+            <>
```
[the remark this answers is not in the document]
Comment (talevy17) [reply packages/plugin-file-upload/web/src/file-upload-viewer.jsx]: done

File: packages/plugin-file-upload/web/src/pdfViewer/pdfIframeSandbox.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (talevy17) [reply packages/plugin-file-upload/web/src/pdfViewer/pdfIframeSandbox.tsx]: done

File: packages/plugin-file-upload/web/src/pdfViewer/pdfViewer.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+export interface PDFViewerProps {
+  status: PDF_STATUS;
+  setStatus: (status: PDF_STATUS) => void;
+  fileData: Record<string, string>;
+  config: PDFConfig;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  scope?: any;
+}
```
[the remark this answers is not in the document]
Comment (talevy17) [reply packages/plugin-file-upload/web/src/pdfViewer/pdfViewer.tsx]: moved to pdfConsts

File: packages/plugin-file-upload/web/src/file-upload-viewer.jsx [hunk 10]
```diff
@@ -282,26 +296,65 @@ class FileUploadViewer extends PureComponent {
...
       >
-        <div className={style} data-hook="fileUploadViewer" ref={this.fileUploadViewerRef}>
-          {viewer}
-          {this.renderAutoDownloadLinkRef()}
-        </div>
+        {shouldRenderPDFViewer ? (
+          <div data-hook={dataHook} ref={this.fileUploadViewerRef}>
+            <PDFViewer
+              status={pdfStatus}
+              dataHook={dataHook}
```
Comment (danarish4) [inline packages/plugin-file-upload/web/src/file-upload-viewer.jsx:340]: not needed

File: packages/ui-components/web/src/Components/SelectionList.tsx [hunk 11]
```diff
@@ -133,16 +134,19 @@ class SelectionList extends Component<SelectionListProps, { focusIndex: number }
       optionClassName,
       useNewSettingsUi,
     } = this.props;
+    const optionsData = dataSource.map(item => this.mapItemToOptionData(item));
+    const subText = (
+      optionsData.filter(optionData => optionData.selected)?.[0].item as Record<string, string>
```
[the remark this answers is not in the document]
Comment (talevy17) [reply packages/ui-components/web/src/Components/SelectionList.tsx:139]: @tamrat-bay great catch, I see you pushed the change, thanks !

**Pull request merged**